packages/cli/src/ui/components/messages/ToolMessage.tsx

```diff
@@ -68,11 +71,61 @@ export const ToolMessage: React.FC<ToolMessageProps> = ({
   const { renderMarkdown } = useUIState();
   const isAlternateBuffer = useAlternateBuffer();
   const isThisShellFocused =
-    (name === SHELL_COMMAND_NAME || name === 'Shell') &&
+    (name === SHELL_COMMAND_NAME ||
+      name === 'Shell' ||
+      name === SHELL_TOOL_NAME) &&
     status === ToolCallStatus.Executing &&
     ptyId === activeShellPtyId &&
     embeddedShellFocused;
 
+  const { setEmbeddedShellFocused } = useUIActions();
+  const containerRef = React.useRef<DOMElement>(null);
+  // The shell is focusable if it's the shell command, it's executing, and the interactive shell is enabled.
+  const isThisShellFocusable =
+    (name === SHELL_COMMAND_NAME ||
+      name === 'Shell' ||
+      name === SHELL_TOOL_NAME) &&
+    status === ToolCallStatus.Executing &&
+    config?.getEnableInteractiveShell();
+
+  useMouse(
```

Comment:
this is worth refactoring into a ShellToolMessage component as all this logic isn't needed for a regular ToolMessage so the non-shell case can be kept a lot lighter weight if we avoid it.